packages/e2e/test/web-extension/dapp-sdk/features/singleUndelegation.ts

```diff
@@ -0,0 +1,32 @@
+import { ObservableWallet } from '@cardano-sdk/wallet';
+import { buildAndSignTx } from '../utils';
+import { combineLatest, take, tap } from 'rxjs';
+
+export const singleUndelegation = ({
+  connectedWallet,
+  logger
+}: {
+  logger: typeof console;
+  connectedWallet: ObservableWallet;
+}) => {
+  const undelegateAssetsElement = document.querySelector('#info-undelegate-assets')!;
+
+  combineLatest([connectedWallet.balance.utxo.available$])
+    .pipe(
+      take(1),
+      tap(async ([availableBalance]) => {
+        if (!connectedWallet) {
+          return logger.warn('Please, connect the wallet first');
+        }
+
+        if (!availableBalance.assets || availableBalance.assets?.size === 0) {
+          throw new Error('Your wallet has no assets');
+        }
+
+        const builtTx = connectedWallet.createTxBuilder().delegatePortfolio(null).build();
+
+        buildAndSignTx({ builtTx, connectedWallet, textElement: undelegateAssetsElement });
```

Comment:
This is a little bit confusing, as we're passing an already 'builtTx' to 'buildAndSignTx'.
```suggestion
        const txBuilder = connectedWallet.createTxBuilder().delegatePortfolio(null);

        buildAndSignTx({ txBuilder, connectedWallet, textElement: undelegateAssetsElement });
```

Reply:
it's addressed here: ac7ca1e3962ca8240a8e701b545712cc53127eaa , although, instead of renaming the txBuilder, I am renaming the function since it's not building but inspecting and signing, to `inspectAndSingTx`